pkg/cfg/config.go

```diff
@@ -54,15 +53,16 @@ type config struct {
 }
 
 var DefaultEnvKeyReplacer = strings.NewReplacer(".", "_", "-", "_")
-var templateRegex = regexp.MustCompile(`{([\w.\-]+)}`)
+var templateRegexp = regexp.MustCompile(`{([\w.\-]+)}`)
+var keyToEnvRegexp = regexp.MustCompile(`\[(\d)\]`)
 
 func New() GosoConf {
-	return NewWithInterfaces(os.LookupEnv)
+	return NewWithInterfaces(NewOsEnvProvider())
 }
 
-func NewWithInterfaces(lookupEnv LookupEnv) GosoConf {
+func NewWithInterfaces(lookupEnv EnvProvider) GosoConf {
```

Comment:
Shouldn't the argument now be called envProvider?

Reply:
renamed